### PR TITLE
Add "debugging" folder/dll to gadgets for gadgets that serve as test/development tools 

### PR DESCRIPTION
**Pull request by apd47**
As mentioned in #1087, this PR moves some of the untested "debugging-type" gadgets into their own subdirectory of "gadgets", and into their own DLL. These gadgets are mostly untested, and shouldn't be used in production pipelines - but rather for debugging purposes by developers/advanced users.

### CODE REVIEW

**Comment by apd47** [comment]
The tests are failing on the ismrmrd_dump_gadget_test.cfg run, but I'm not sure why. That test uses a different format, where instead of a configuration being set, a "template" is used instead:
template=ismrmrd_dump_gadget_test.template.xml

The XML file it specifies doesn't seem to exist either - does anyone happen to know how this test works so I can chase down the failure?

**Comment by apd47** [comment]
@hansenms - looks like moving the dump gadget back into mri_core (where it should be) means test pass - I'll still take a look at the template method setup for next time, but I think we're set on this one whenever you get a chance.

File: gadgets/mri_core/CMakeLists.txt [hunk 1]
```diff
@@ -68,11 +62,10 @@ set(gadgetron_mricore_header_files GadgetMRIHeaders.h
         generic_recon_gadgets/GenericImageReconArrayToImageGadget.h
         generic_recon_gadgets/GenericReconReferenceKSpaceDelayedBufferGadget.h
         generic_recon_gadgets/GenericReconImageToImageArrayGadget.h 
-        WhiteNoiseInjectorGadget.h
         ImageFinishGadget.h
         dependencyquery/NoiseSummaryGadget.h
         NHLBICompression.h
-    	cpuisa.h
+    	  cpuisa.h
```
Comment (hansenms) [inline gadgets/mri_core/CMakeLists.txt:68]: nit: extra space here

**Comment by hansenms** [comment]
I think we need to validate if any of these gadgets are used in any chains and update those. See for example: https://github.com/gadgetron/gadgetron/blob/master/gadgets/python/legacy/config/pseudoreplica.xml

**Comment by apd47** [comment]
Oh - good catch. I'd checked tests but not all the untested configs. Working through now.

**Comment by apd47** [comment]
My search through all the configs looks like Pseudoreplica.xml was the only one using any of these 6 gadgets, so that's been updated. Also fixed the extra space in cmakelists

**Comment by apd47** [comment]
(accidental close - tabbed to the wrong button...)

**Review by hansenms — CHANGES_REQUESTED** [review]
Can we bump the patch version at least just to make it clear that things are moving around.

**Comment by apd47** [comment]
Sounds like a good idea - I'm not sure how to do that, actually

**Comment by hansenms** [comment]
@apd47 version numbers are in the main CMakeLists.txt file.

**Review by hansenms — APPROVED** [review]

**Pull request merged**